File: apis/ocpdiag/ocpdiag/core/params/utils_test.cc
// Copyright 2022 Google LLC
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file or at
// https://opensource.org/licenses/MIT.

#include "ocpdiag/core/params/utils.h"

#include "gmock/gmock.h"
#include "gtest/gtest.h"
#include "absl/status/status.h"
#include "ocpdiag/core/params/testdata/test_params.pb.h"
#include "ocpdiag/core/testing/file_utils.h"
#include "ocpdiag/core/testing/status_matchers.h"

namespace {

using ::ocpdiag::params::JsonFileToMessage;
using ::ocpdiag::testing::StatusIs;
using ::ocpdiag::core::params::testdata::Params;

TEST(JsonFileToMessageTest, BadFile) {
  Params params;
  EXPECT_THAT(JsonFileToMessage("/not/a/valid/path", &params),
              StatusIs(absl::StatusCode::kNotFound));
}

TEST(JsonFileToMessageTest, BadMessage) {
  Params params;
  std::string test_path = ocpdiag::testutils::GetDataDependencyFilepath(
      "ocpdiag/core/params/testdata/bad_test_params.json");
  EXPECT_THAT(JsonFileToMessage(test_path.c_str(), &params),
              StatusIs(absl::StatusCode::kInvalidArgument));
}

TEST(JsonFileToMessageTest, GoodMessage) {
  Params params;
  std::string test_path = ocpdiag::testutils::GetDataDependencyFilepath(
      "ocpdiag/core/params/testdata/test_params.json");
  ASSERT_OK(JsonFileToMessage(test_path.c_str(), &params));
  EXPECT_EQ(params.foo(), "foo_string");
}

}  // namespace

File: apis/ocpdiag/ocpdiag/core/params/utils_test.py
# Copyright 2022 Google LLC
#
# Use of this source code is governed by an MIT-style
# license that can be found in the LICENSE file or at
# https://opensource.org/licenses/MIT.

"""Tests for ocpdiag.core.params.utils."""

import io
import sys

import unittest
from ocpdiag.core.params import utils
from ocpdiag.core.params.testdata import test_params_pb2


class UtilsTest(unittest.TestCase):

  def test_json_parses_correctly(self):
    sys.stdin = io.StringIO("""{
      "foo" : "foovalue",
      "bar" : 42,
      "msg" : {
        "sub" : "submessage",
        "other" : [1, 2, 3]
      }
    }""")
    msg = utils.GetParams(test_params_pb2.Params())
    self.assertEqual(msg.foo, "foovalue")
    self.assertEqual(msg.bar, 42)
    self.assertEqual(msg.msg.sub, "submessage")
    self.assertEqual(msg.msg.other, [1, 2, 3])


if __name__ == "__main__":
  unittest.main()
